### PR TITLE
fix(react-native): improve error logging for speaker manager hook and improve usage of incall manager in SDK

### PR DESCRIPTION
**Pull request by khushal87**
The PR addresses the following things:
- Improve error logging for speaker management in client and link to appropriate docs.
- Fix RN incall manager usage in SDK.
- Improve speaker manager docs in the SDK.

### CODE REVIEW

File: packages/client/src/devices/InputMediaDeviceManager.ts [hunk 1]
```diff
@@ -130,7 +130,9 @@ export abstract class InputMediaDeviceManager<
    */
   async select(deviceId: string | undefined) {
     if (isReactNative()) {
-      throw new Error('This method is not supported in React Native');
+      throw new Error(
+        'This method is not supported in React Native. Please visit http://localhost:3000/chat/docs/sdk/reactnative/core/camera-and-microphone/#speaker-management for reference.',
```
Comment (oliverlaz) [inline packages/client/src/devices/InputMediaDeviceManager.ts:134]:
```suggestion
        'This method is not supported in React Native. Please visit https://getstream.io/video/docs/reactnative/core/camera-and-microphone/#speaker-management for reference.',
```

We shouldn't have localhost in the error message :)

File: packages/client/src/devices/SpeakerManager.ts [hunk 2]
```diff
@@ -38,6 +38,11 @@ export class SpeakerManager {
    * @returns an Observable that will be updated if a device is connected or disconnected
    */
   listDevices() {
+    if (isReactNative()) {
```
Comment (oliverlaz) [inline packages/client/src/devices/SpeakerManager.ts:41]: Can you also add a note in the JS doc similar like for `.select()` as well?

File: packages/client/src/devices/InputMediaDeviceManager.ts (hunk 1 above)
Comment (khushal87) [reply packages/client/src/devices/InputMediaDeviceManager.ts:134]: Oops, I added this by mistake. Let me fix it.

Comment (khushal87) [reply packages/client/src/devices/InputMediaDeviceManager.ts]: Fixed 👍🏻

File: packages/client/src/devices/SpeakerManager.ts (hunk 2 above)
Comment (khushal87) [reply packages/client/src/devices/SpeakerManager.ts:42]: Fixed 👍🏻

File: packages/react-native-sdk/docusaurus/docs/reactnative/03-core/04-camera-and-microphone.mdx [hunk 3]
```diff
@@ -139,27 +139,48 @@ const { status } = useMicrophoneState(); // status returns enabled, disabled or
...
 
-InCallManager allows you to enable/disable the speaker phone.
+InCallManager doesn't allow you to change the output device if you have multiple devices. However, you can turn on the speaker and route the audio through speaker, using:
 
 ```tsx
 import InCallManager from 'react-native-incall-manager';
 
 // For Android
-InCallManager.setSpeakerphoneOn(enable: ?boolean); // Pass true for on, and false for off.
+InCallManager.setSpeakerphoneOn(true); // Pass true for on, and false for off.
```
Comment (santhoshvai) [inline packages/react-native-sdk/docusaurus/docs/reactnative/03-core/04-camera-and-microphone.mdx:180]: have you tested this on android?? 

I would say not to recommend this method at all

File: packages/react-native-sdk/docusaurus/docs/reactnative/03-core/04-camera-and-microphone.mdx [hunk 4]
```diff
@@ -139,27 +139,48 @@ const { status } = useMicrophoneState(); // status returns enabled, disabled or
...
 
 // To force speaker phone on/off on both android and iOS.
-InCallManager.setForceSpeakerphoneOn(flag: ?boolean) // Pass true for on, and false for off.
+InCallManager.setForceSpeakerphoneOn(true); // Pass true for on, and false for off. Once off audio is always routed through earpiece.
 ```
```
Comment (santhoshvai) [inline packages/react-native-sdk/docusaurus/docs/reactnative/03-core/04-camera-and-microphone.mdx:184]: `Once off audio is always routed through earpiece.`

have you tested on android as well?

**Review by santhoshvai — APPROVED** [review]

File: sample-apps/react-native/dogfood/src/screens/AudioRoom/Room.tsx [hunk 5]
```diff
@@ -18,6 +19,12 @@ export default function Room({ onClose }: { onClose: () => void }) {
   const callingState = useCallCallingState();
   const call = useCall();
 
+  // Automatically route audio to speaker devices as relevant for watching videos.
+  useEffect(() => {
+    InCallManager.start({ media: 'audio' });
+    return () => InCallManager.stop();
+  }, []);
```
Comment (santhoshvai) [inline sample-apps/react-native/dogfood/src/screens/AudioRoom/Room.tsx:26]: comment says route to speaker.. but media is audio..

Comment (khushal87) [reply sample-apps/react-native/dogfood/src/screens/AudioRoom/Room.tsx:26]: Let me fix this

**Pull request merged**